solutions/usaco-939.mdx

```diff
@@ -7,10 +7,81 @@ author: Maggie Liu
 
 [Official Analysis](http://www.usaco.org/current/data/sol_buckets_bronze_open19.html)
 
+## Explanation
+
+If there was no rock, then the number of cows needed would just be the
+horizontal distance between the barn and the lake, plus the vertical distance
+between the barn and the lake, minus $1$. Adding the rock onto the farm usually
+doesn't change the answer. The only case where the rock matters is if the rock,
+barn and lake are on the same line, and the rock is between the barn and the
+lake. In this case, there will need to be $2$ extra cows to go around the rock.
+
 ## Implementation
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cstdio>
+using namespace std;
+
+int main()
+{
+	freopen("buckets.in", "r", stdin);
+	freopen("buckets.out", "w", stdout);
+	int barn_i = 0, barn_j = 0, rock_i = 0, rock_j = 0, lake_i = 0, lake_j = 0;
+	for (int i = 0; i < 10; i++)
+	{
+		string row;
+		cin >> row;
+		for (int j = 0; j < 10; j++)
+		{
+			if (row[j] == 'B')
+			{
+				barn_i = i;
+				barn_j = j;
+			}
+			else if (row[j] == 'R')
+			{
+				rock_i = i;
+				rock_j = j;
+			}
+			else if (row[j] == 'L')
+			{
+				lake_i = i;
+				lake_j = j;
+			}
+		}
+	}
+
+	// initial distance
+	int cows = abs(barn_i - lake_i) + abs(barn_j - lake_j) - 1;
+
+	// if the barn, lake and rock are in the same column
+	// and the rock is betwen the barn and the lake
+	if (barn_i == lake_i && rock_i == barn_i
+			&& ((lake_j < rock_j && rock_j < barn_j)
+					|| (barn_j < rock_j && rock_j < lake_j)))
```

Comment:
think the third line of the if statement should be on the same indentation level as the second line

Reply:
but should't it be indented more since the or expression is inside the entire and expression?
```
if (a && (b || c))

if (a
	&& (b
		|| c))

Reply:
then uh smth like this
```cpp
	else if (barn_j == lake_j && rock_j == barn_j
				&& ((lake_i < rock_i && rock_i < barn_i)
				    || (barn_i < rock_i && rock_i < lake_i)))
```

Reply:
this
![image](https://user-images.githubusercontent.com/63619830/124392458-8cd12580-dcaa-11eb-8ae6-116fd93f1864.png)
or this?
![image](https://user-images.githubusercontent.com/63619830/124392443-7f1ba000-dcaa-11eb-89a8-7b54ca3147dd.png)

Reply:
and i thought the or should have 1 more indent than the parentheses?

Reply:
first one looks better tbh